### PR TITLE
Make equilibration cache files more human readable

### PR DESCRIPTION
**Pull request by badisa**
* Allows running multiple inference runs in the same dir without
  colliding

### CODE REVIEW

File: examples/validate_relative_binding.py [hunk 1]
```diff
@@ -45,6 +45,26 @@
 from md import builders, minimizer
 from rdkit.Chem import rdFMCS
 
+
+def cache_wrapper(cache_path: str, f):
```
Comment (badisa) [inline examples/validate_relative_binding.py:49]: Feels like we might want to put this somewhere else so we can use this sort of logic in other scripts.

Comment (maxentile) [inline examples/validate_relative_binding.py:49]: At first glance I thought `f` was a file object.

Naming suggestions:
* retain single letter name, add type hint `(cache_path: str, f)` -> `(cache_path: str, f: callable)`
* expand single letter name, so type hint would be redundant `(cache_path: str, f)` -> `(cache_path: str, fxn)`

File: examples/validate_relative_binding.py [hunk 2]
```diff
@@ -181,28 +201,24 @@
     pressure = 1.0
     dt = 2.5e-3
 
+    cached_equil_path = "equil_{}_blocker_{}.pkl".format(
+        os.path.basename(cmd_args.protein_pdb).split(".")[0].replace(" ", "_"), cmd_args.blocker_name.replace(" ", "_")
```
Comment (maxentile) [inline examples/validate_relative_binding.py:205]: Seems a safe assumption that no directory names in path/to/proteinpdb contain a " ", not sure if we might want to do any other string -> filename sanitization here

File: examples/validate_relative_binding.py [hunk 3]
```diff
@@ -45,6 +45,26 @@
 from md import builders, minimizer
 from rdkit.Chem import rdFMCS
 
+
+def cache_wrapper(cache_path: str, f):
+    """Given a path and a function, will either write the result of the function call
+    to the file or read form the file. The output of the function is expected to be pickable
```
Comment (maxentile) [inline examples/validate_relative_binding.py:51]: Spelling

File: examples/validate_relative_binding.py (hunk 1 above)
Comment (maxentile) [inline examples/validate_relative_binding.py:49]: Should we add an optional argument `overwrite=False`? (Perhaps a brief comment about when and how we should invalidate the cache would be helpful too!)

**Review by maxentile — APPROVED** [review]
Looks good!

**Pull request merged**